### PR TITLE
Try to fix #36.

### LINKED ISSUE #36: Error building with glibc 2.33 because of _STAT_VER changes
The glibc maintainers have done something with the `stat` wrappers in 2.33: https://github.com/bminor/glibc/commit/8ed005daf0ab03e142500324a34087ce179ae78e

While running `make` to build `ekam` under glibc 2.33:
```
✘ learn: ekam/rules/intercept.ekam-rule
    src/ekam/rules/intercept.c: In function ‘direct_stat’:
    src/ekam/rules/intercept.c:671:21: error: ‘_STAT_VER’ undeclared (first use in this function)
      671 |   return real_xstat(_STAT_VER, path, sb);
          |                     ^~~~~~~~~
    src/ekam/rules/intercept.c:671:21: note: each undeclared identifier is reported only once for each function it appears in
    src/ekam/rules/intercept.c:672:1: warning: control reaches end of non-void function [-Wreturn-type]
      672 | }
          | ^
```

The `intercept.c` line in question: https://github.com/capnproto/ekam/blob/454152e658db071f62a45eae620ef244c1726165/src/ekam/rules/intercept.c#L671

Note that `ekam` previously built on the same Linux install but a glibc upgrade broke compilation as seen above, so I suspect this issue is coming to Debian/Ubuntu eventually.

For now, I built an `ekam` binary in a Docker container using an older glibc, and I replaced the `_STAT_VER` in my host system's `intercept.c` with `3` and things seem to work. I figured I'd put this up now for anyone else Googling. I'm a bit out of my depth on what's going on in that glibc change, or why `_STAT_VER` didn't make it into _anything_ under my `/usr/include` (I have no `xstatver.h`, for example), but I'll try to poke at it some more in a bit.

### PR DESCRIPTION
**Pull request by kentonv**
@bretthoerner Can you try this and let me know if it works?

### CODE REVIEW

File: src/ekam/rules/intercept.c [hunk 1]
```diff
@@ -669,6 +708,18 @@ static int direct_stat(const char* path, struct stat* sb) {
   }
 
   return real_xstat(_STAT_VER, path, sb);
+#else  /* defined(_STAT_VER) */
+  // glibc 2.33+ no longer defines stat() as an inline wrapper in the header, but instead exports
+  // it as a regular symbol.
+  static stat_t* real_stat = NULL;
+
+  if (real_stat == NULL) {
+    real_stat = (xstat_t*) dlsym(RTLD_NEXT, "stat");
```
Comment (bretthoerner) [inline src/ekam/rules/intercept.c:717]: It worked with `s/xstat_t/stat_t/`, without:

```
✘ learn: ekam/rules/intercept.ekam-rule
    src/ekam/rules/intercept.c: In function ‘direct_stat’:
    src/ekam/rules/intercept.c:717:18: error: ‘xstat_t’ undeclared (first use in this function); did you mean ‘
      lstat_t’?
      717 |     real_stat = (xstat_t*) dlsym(RTLD_NEXT, "stat");
          |                  ^~~~~~~
          |                  lstat_t
    src/ekam/rules/intercept.c:717:18: note: each undeclared identifier is reported only once for each function it
      appears in
    src/ekam/rules/intercept.c:717:26: error: expected expression before ‘)’ token
      717 |     real_stat = (xstat_t*) dlsym(RTLD_NEXT, "stat");
          |                          ^
```

Comment (bretthoerner) [inline src/ekam/rules/intercept.c:717]:
```suggestion
    real_stat = (stat_t*) dlsym(RTLD_NEXT, "stat");
```

**Comment by kentonv** [comment]
Thanks, I was pretty sure I had at least one typo like that. :)

Applied your fixed, squashed, added a proper commit message.

**Review by bretthoerner — APPROVED** [review]
```
=====================================================
SUCCESS
Output is at bin/ekam
=====================================================
```

Thanks!

**Pull request merged**